app/src/main/java/com/example/mystore/ui/home/layouts/ProductRow.kt

```diff
@@ -0,0 +1,102 @@
+package com.example.mystore.ui.home.layouts
+
+
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.Divider
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color.Companion.Black
+import androidx.compose.ui.res.dimensionResource
+import androidx.compose.ui.text.font.FontWeight
+import coil.compose.rememberAsyncImagePainter
+import com.example.mystore.R
+import com.example.mystore.model.ProductModel
+import com.example.mystore.ui.products.layouts.RatingBar
+import com.example.mystore.ui.theme.Gray
+import com.example.mystore.ui.theme.LightGray40
+import java.text.NumberFormat
+import java.util.Locale
+
+@Composable
+fun ProductRow(
+    product: ProductModel,
+    navigateToProductDetails: (productId: Int) -> Unit = {}
+) {
+    Column {
+        Row(
+            modifier = Modifier
+                .clickable { navigateToProductDetails(product.id) },
+            verticalAlignment = Alignment.CenterVertically,
+            horizontalArrangement = Arrangement.SpaceBetween
+        ) {
+
+            Box(modifier = Modifier.width(dimensionResource(id = R.dimen.product_image_size))) {
```

Comment:
Unnecessary wrapper, you can set the width on the image the same way you've set the height. Is there something I'm missing?